### PR TITLE
feat(shorebird_cli): write all stdout and stderr to log file

### LINKED ISSUE #2147: fix: logger progress, confirm, prompt, etc. output not captured in log files
`ShorebirdLogger` writes info, detail, error, etc to a log file, but output from prompts, confirms, progress, etc. are not captured. The logs would be more helpful if they were an exact copy of what the user sees when running a command with the -v flag.

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Creates a `Stdout` wrapper class that writes all stdout and stderr output to a log file in addition to stdout and stderr.

Fixes https://github.com/shorebirdtech/shorebird/issues/2147

Sample output from `shorebird init`:

```
2024-05-30T13:16:57.330628 [Process.run] /Users/bryanoltman/Documents/sandbox/logs_test/android/gradlew app:tasks --all --console=auto (in /Users/bryanoltman/Documents/sandbox/logs_test/android)
2024-05-30T13:16:57.334988 [Process.run] xcodebuild -list (in /Users/bryanoltman/Documents/sandbox/logs_test/ios)
2024-05-30T13:16:57.412013 
⠙ Detecting product flavors... (83ms)
2024-05-30T13:16:57.491825 
⠹ Detecting product flavors... (0.2s)
2024-05-30T13:16:57.571239 
⠸ Detecting product flavors... (0.2s)
2024-05-30T13:16:57.650606 
⠼ Detecting product flavors... (0.3s)
2024-05-30T13:16:57.730988 
⠴ Detecting product flavors... (0.4s)
2024-05-30T13:16:57.810432 
⠦ Detecting product flavors... (0.5s)
2024-05-30T13:16:57.890542 
⠧ Detecting product flavors... (0.6s)
2024-05-30T13:16:57.970921 
⠇ Detecting product flavors... (0.6s)
2024-05-30T13:16:58.050298 
⠏ Detecting product flavors... (0.7s)
2024-05-30T13:16:58.118876 Exited with code 0
2024-05-30T13:16:58.119232 
stdout:
> Task :gradle:compileJava NO-SOURCE
> Task :gradle:compileGroovy UP-TO-DATE
> Task :gradle:pluginDescriptors UP-TO-DATE
> Task :gradle:processResources UP-TO-DATE
> Task :gradle:classes UP-TO-DATE
> Task :gradle:jar UP-TO-DATE

> Task :app:tasks
```

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [ ] 🧪 Tests

### CODE REVIEW

**Comment by bryanoltman** [comment]
TODO: maintain verbose logs even when verbose logging is not on (probably resurrect parts of ShorebirdLogger that shouldn't have been deleted)

File: packages/shorebird_cli/test/src/extensions/string_test.dart [hunk 1]
```diff
@@ -1,16 +1,61 @@
...
+    test('returns string unchanged if no ansi escape codes are present', () {
+      expect('test'.removeAnsiEscapes(), equals('test'));
+    });
   });
 
-  test('IsUpperCase', () async {
-    expect('TEST'.isUpperCase(), isTrue);
-    expect('test'.isUpperCase(), isFalse);
-    expect('Test'.isUpperCase(), isFalse);
+  group('IsUpperCase', () {
```
Comment (felangel) [inline packages/shorebird_cli/test/src/extensions/string_test.dart:54]:
```suggestion
  group('isUpperCase', () {
```

File: packages/shorebird_cli/test/src/extensions/string_test.dart [hunk 2]
```diff
@@ -1,16 +1,61 @@
+import 'package:mason_logger/mason_logger.dart';
 import 'package:shorebird_cli/src/extensions/string.dart';
 import 'package:test/test.dart';
 
 void main() {
-  test('IsNullOrEmpty', () async {
-    expect(null.isNullOrEmpty, isTrue);
-    expect(''.isNullOrEmpty, isTrue);
-    expect('test'.isNullOrEmpty, isFalse);
+  group('IsNullOrEmpty', () {
```
Comment (felangel) [inline packages/shorebird_cli/test/src/extensions/string_test.dart]:
```suggestion
  group('isNullOrEmpty', () {
```

**Review by felangel — APPROVED** [review]

File: packages/shorebird_cli/test/src/extensions/string_test.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/extensions/string_test.dart:54]: These groups are named after the extension names, in this case:

```dart

extension IsUpperCase on String {
  /// Returns `true` if this string is in uppercase.
  bool isUpperCase() => this == toUpperCase();
}
```

File: packages/shorebird_cli/test/src/extensions/string_test.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/extensions/string_test.dart]: Corrected this to `NullOrEmpty` to reflect the extension name.

**Pull request merged**